bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java

```diff
@@ -3007,6 +3007,14 @@ Point mapInPixels (Control from, Control to, int x, int y) {
 	return new Point (point.x, point.y);
 }
 
+private int getZoomLevelForMapping(Control from, Control to) {
+	return Optional.ofNullable(to)
+            .map(Control::getZoom)
+            .orElseGet(() -> Optional.ofNullable(from)
+                    .map(Control::getZoom)
+                    .orElse(0));
```

Comment:
Two question on this:
1. Does it make sense to default the zoom value to "0"? In case this really happens, the results will be completely unusable, so I assume you do not expect this case to occur at all. Wouldn't it be better to thrown an exception then to make explicit what goes wrong if that case really happens?
2. I am not sure whether using stream operations is a good idea here. It produces quite some overhead and the "simple" implementation would, in my opinion, be even easier to understand, as would look something like this, wouldn't it:
```java
if (to != null) {
	return to.getZoom();
}
return from.getZoom();
```

Reply:
Adapted as requested